app/src/main/java/unipd/se18/ocrcamera/SearchResultActivity.java

```diff
@@ -0,0 +1,407 @@
+package unipd.se18.ocrcamera;
+
+import android.app.SearchManager;
+import android.content.Intent;
+import android.os.Bundle;
+import android.os.Process;
+import android.support.annotation.NonNull;
+import android.support.v4.app.FragmentManager;
+import android.support.v7.app.AppCompatActivity;
+import android.util.Log;
+import android.view.KeyEvent;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.AutoCompleteTextView;
+import android.widget.Button;
+import android.widget.Filter;
+import android.widget.ListView;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import unipd.se18.ocrcamera.inci.Ingredient;
+import unipd.se18.ocrcamera.inci.IngredientsExtractor;
+
+
+/**
+ * This activity allow the user to manually search for ingredients, providing smart suggestions
+ * while the user is typing.
+ * Currently it is launched with an ACTION_SEARCH intent from the SearchView contained in the
+ * menu, inside ResultActivity.java
+ * @author Luca Moroldo g3
+ */
+public class SearchResultsActivity extends AppCompatActivity {
+
```

Comment:
I would have preferred a comment to know what this variable (TAG) is for

Reply:
It's a variable used just to save the activity name and easily bog errors / debug / info messages, you can find this variable in any class of the application.